### PR TITLE
Remove Object#namespace

### PR DESCRIPTION
**Pull request by bdunne**
Based on feedback

### CODE REVIEW

File: spec/core_ext/object/namespace_spec.rb [hunk 1]
```diff
@@ -19,26 +19,8 @@ module Eee; end
       expect(Aaa::Bbb::Ccc::Eee.in_namespace?("Aaa::Bbb")).to be_truthy
     end
 
-    it "Instance of Class" do
-      expect(Aaa::Bbb::Ccc::Ddd.new.in_namespace?(Aaa::Bbb)).to be_truthy
-    end
-
```
Comment (matthewd) [inline spec/core_ext/object/namespace_spec.rb]: Shouldn't `Object#in_namespace?` go away too? It's now broken (per this test removal), right?

Comment (brandondunne) [inline spec/core_ext/object/namespace_spec.rb]: It still works for the class or module itself, just not an instance

Comment (matthewd) [inline spec/core_ext/object/namespace_spec.rb]: Right.. so we should move it to Module

Comment (brandondunne) [inline spec/core_ext/object/namespace_spec.rb]: @matthewd Updated to work for instances as well.

**Comment by matthewd** [comment]
I'm not convinced `in_namespace?` makes any more sense as a global/Object method than `namespace` does... but I guess it's at least a slightly more obscure name.

¯\\\_(ツ)_/¯

**Comment by brandondunne** [comment]
@Fryguy Any thoughts?

**Comment by brandondunne** [comment]
bump :turtle:

**Pull request merged**